include/turtle/detail/function_impl_template.hpp

```diff
@@ -97,40 +97,52 @@ namespace detail
 
         struct wrapper : wrapper_base< R, expectation_type >
         {
+        private:
+            typedef wrapper_base< R, expectation_type > Base;
```

Comment:
This needs to be `base_type` instead of `Base` if I correctly remember the coding conventions.

Reply:
Not sure the typedef brings much, but it should probably be used consistently then (in the other constructor)